### ISSUE TITLE
Add TypeScript support

### ISSUE THREAD

**Issue by vahdet** [JamesBrill/react-speech-recognition#51]
Firstly, thanks for the promising work!
In my case, the basic import (`import SpeechRecognition, { useSpeechRecognition } from 'react-speech-recognition'`) fails the TypeScript check with code *ts(7016)*:
```
Could not find a declaration file for module 'react-speech-recognition'. '.../my-proj/node_modules/react-speech-recognition/lib/index.js' implicitly has an 'any' type.
  Try `npm install @types/react-speech-recognition` if it exists or add a new declaration (.d.ts) file containing `declare module 'react-speech-recognition';`
```

And `npm install @types/react-speech-recognition` throws:

```
Not Found - GET https://registry.npmjs.org/@types%2freact-speech-recognition - Not found
```

So, I had to workaround it with adding this in my `app.d.ts file`:

```
declare module 'react-speech-recognition';
```

So, do you have any agenda for adding TypeScript support? If needed, I also may contribute at my best effort.

**Comment by JamesBrill** [comment]
Hi @vahdet thanks for the suggestion. I have no experience with TypeScript (though it is of interest to me), so you may be better positioned to contribute a declaration file than me. If you make a PR for this, I can advise on the types and shape of the inputs/outputs.

You probably don't even need to do the whole thing - if you get it started and help me get an idea of how to structure and publish a declaration file, I can probably fill in the missing definitions.

**Comment by vahdet** [comment]
@JamesBrill  Well, yesterday I tinkered with the code indeed. I forked the repo applied my work on a new branch [`feature/typecript-support`](https://github.com/vahdet/react-speech-recognition/tree/feature/typescript-support). I have some experience with TypeScript, but cannot call myself a master of it. Anyways, I can send a PR w/ my further explanation on what I've done. And then, improvements can be done on it. I won't be around for a couple of weeks, so I may react slower than today..
